open-api/rest-catalog-open-api.yaml

```diff
@@ -2893,6 +3003,37 @@ components:
           additionalProperties:
             type: string
 
+    AppendDataFileRequest:
+      type: object
+      required:
+        - data-files
+      properties:
+        identifier:
+          allOf:
+            - $ref: '#/components/schemas/TableIdentifier'
+          description: Table identifier to update; must be present for CommitTransactionRequest
+        accept-delay-ms:
+          type: integer
+          format: int64
+          description: Acceptable delay in milliseconds for processing the append request.
+          example: 300000
+        data-files:
+          type: array
+          items:
+            $ref: '#/components/schemas/DataFile'
+          description: A list of serialized data files to append to the table metadata.
+
+    AppendDataFileResponse:
+      type:
+        object
+      properties:
+        location:
+          type: string
+          description:
+            URL to monitor the status of the append data file operation. This facilitates tracking the progress and
```

Comment:
I am not sure every REST catalog will ready to provide a URL to visualize the append progress. Also for async operation like this, how many snapshot would be generated, how do we cancel or abort such operation?